### PR TITLE
docs: Add README for influxdb_client

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
In order to encourage more contributions (like the great one from @aakashhemadri https://github.com/influxdata/influxdb_iox/pull/988) we should be clearer about the goals and expectations for the influxdb_iox client. 

# Changes
Add a README that explains what `influxdb_iox` client is and our goals

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: influxdb2_client/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# InfluxDB V2 Client API
+
+This crate contains a work-in-progress implementation of a Rust client for the [InfluxDB 2.0 API](https://docs.influxdata.com/influxdb/v2.0/reference/api/).
+
```
Comment (pauldix) [inline influxdb2_client/README.md:4]:
```suggestion

This client is not the Rust client for IOx. You can find that [here](../influxdb_iox_client).
```

**Review by pauldix — APPROVED** [review]
Just one suggested addition

File: influxdb2_client/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## Contributing
+
+If you would like to contribute code you can do through GitHub by forking the repository and sending a pull request into the master branch.
+
+
+## Future work
+
+- [ ] Publish as a crate on [crates.io](http://crates.io)
+
+If you would like to contribute code you can do through GitHub by forking the repository and sending a pull request into the master branch.
```
Comment (tustvold) [inline influxdb2_client/README.md:22]:
```suggestion
If you would like to contribute code you can do through GitHub by forking the repository and sending a pull request into the main branch.
```

File: influxdb2_client/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# InfluxDB V2 Client API
+
+This crate contains a work-in-progress implementation of a Rust client for the [InfluxDB 2.0 API](https://docs.influxdata.com/influxdb/v2.0/reference/api/).
+
+This client is not the Rust client for IOx. You can find that [here](../influxdb_iox_client).
+The InfluxDB IOx project plans to focus its efforts on the subset of the API which are most relevent to IOx, but we accept (welcome!) PRs for adding the other pieces of functionality.
```
Comment (alamb) [inline influxdb2_client/README.md:6]:
```suggestion

The InfluxDB IOx project plans to focus its efforts on the subset of the API which are most relevent to IOx, but we accept (welcome!) PRs for adding the other pieces of functionality.
```

**Pull request merged**